File: src/com/boardgamegeek/provider/BuddiesIdProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.Buddies;
import com.boardgamegeek.util.SelectionBuilder;

public class BuddiesIdProvider extends BaseProvider {
	BuddiesProvider mProvider = new BuddiesProvider();

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		int buddyId = Buddies.getBuddyId(uri);
		return mProvider.buildSimpleSelection(uri).whereEquals(Buddies.BUDDY_ID, buddyId);
	}

	@Override
	protected String getPath() {
		return addIdToPath(mProvider.getPath());
	}

	@Override
	protected String getType(Uri uri) {
		return Buddies.CONTENT_ITEM_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/BuddiesIdProvider.kt
package com.boardgamegeek.provider

import android.net.Uri
import com.boardgamegeek.provider.BggContract.Buddies
import com.boardgamegeek.provider.BggContract.Companion.PATH_BUDDIES
import com.boardgamegeek.provider.BggDatabase.Tables

class BuddiesIdProvider : BaseProvider() {
    override fun getType(uri: Uri) = Buddies.CONTENT_ITEM_TYPE

    override val path: String = "$PATH_BUDDIES/*"

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val buddyName = Buddies.getBuddyName(uri)
        return SelectionBuilder()
            .table(Tables.BUDDIES)
            .whereEquals(Buddies.Columns.BUDDY_NAME, buddyName)
    }
}
